workflow/controller/operator.go

```diff
@@ -1524,6 +1524,10 @@ func (woc *wfOperationCtx) inferFailedReason(pod *apiv1.Pod, tmpl *wfv1.Template
 	sort.Slice(ctrs, func(i, j int) bool { return order(ctrs[i].Name) < order(ctrs[j].Name) })
 	// Init containers have the highest preferences over other containers.
 	ctrs = append(pod.Status.InitContainerStatuses, ctrs...)
+	// When there isn't any containstatus (such as no stock in public cloud), return Failure.
+	if len(ctrs) == 0 {
+		return wfv1.NodeFailed, ""
```

Comment:
We should fill out the message (second return parameter) here, to pass something to the user. Otherwise they're going to struggle to determine why their node has failed.

Reply:
ok, I will fill out the message.